### ISSUE TITLE
Rule for assert statements that are always true

### ISSUE THREAD

**Issue by ValentinCalomme** [astral-sh/ruff#9750]
I have been searching the documentation, but I cannot seem to find a rule that checks for "tautologies" in assert statements. I recently had a case in my codebase where I had an assert statement in the form of:

```py
assert my_dict["key"] == my_dict["key"]
```

Whereas I wanted to have

```py
assert my_dict["key"] == expected_dict["key"]
```

A rule that would check for statements that are always true would be helpful, at the least for cases where we have an equality check of an item with itself.

Does such a rule exist? Should it be added?

**Comment by akx** [comment]
Sibling rules:

* [assert-tuple](https://docs.astral.sh/ruff/rules/assert-tuple/)
* [pytest-assert-always-false](https://docs.astral.sh/ruff/rules/pytest-assert-always-false/) 
* [assert-on-string-literal](https://docs.astral.sh/ruff/rules/assert-on-string-literal/)
